pkg/apis/db/v1alpha1/cassandracluster_types.go

```diff
@@ -93,6 +94,31 @@ const (
 	ContinueResyncLoop = false
 )
 
+func index(slice []string, item string) int {
+	for i := range slice {
+		if slice[i] == item {
+			return i
+		}
+	}
+	return -1
+}
+
+// ClusterPhaseVal returns corresponding float64 value
+func ClusterPhaseVal(clusterPhase string) float64 {
+	return float64(
+		index([]string{ClusterPhaseInitial, ClusterPhaseRunning, ClusterPhasePending}, clusterPhase),
```

Comment:
As discussed, I think that this approach could lead to a lot of mistakes, due to implicit link between the `ClusterPhase` and the `float` returned. 
It could be have at least two other solutions : 

1. To expose the link using a struct, which will avoid mistakes, when we will introduce new ClusterPhase : 

```go
type ClusterPhase struct {
  Id float `json:"clusterId"`
  Name string `json:name`
}
```

2. To define one metric per ClusterPhase, using a boolean to specify if the cluster is active or not in this phase.

Reply:
Yes I think we agree. I wanted to avoid having to duplicate code but having a place in the code where we can see the actual value definitely makes sense.
I don't understand your second point.

Reply:
@erdrix why do you use marshable syntax in the type definition ? Because of the second point ?

Reply:
The second point was to have for each ClusterAction / ClusterPhase type a boolean metric. 
For the marshable syntax, I make a quick copy/paste and forgot to remove it ^^